pkg/p2p/libp2p/connections_test.go

```diff
@@ -252,8 +215,15 @@ func TestDoubleConnectOnAllAddresses(t *testing.T) {
 		expectPeers(t, s2, overlay1)
 		expectPeersEventually(t, s1, overlay2)
 
-		if _, err := s2.Connect(ctx, addr); err == nil {
-			t.Fatal("second connect attempt should result with an error")
+		if _, err := s2.Connect(ctx, addr); !errors.Is(err, p2p.ErrAlreadyConnected) {
```

Comment:
so, the test says that it connects to all addresses which is not correct. it just connects to 1 address at a time. i would expect it to try to connect on all addresses (although i'm not sure which benefit it brings, except making sure the libp2p binds to the appropriate addresses, ports and protocols as expected from configuration, which should be tested as part of the libp2p abstraction, not as part of connection logic), or at least to try repeated connections and testing that the connect-disconnect logic that throws the peer out works as expected.

Reply:
This is the old tests, I just changed the connection state after double connect.  @janos I believe you did the test, so could you comment pls?

Reply:
Yes it connects to all addresses, but one at the time. A parallel test would be a different one. This one ensures that everything is cleaned up on disconnects related to different peers. @acud If you think that this test is not good, feel free to address it in another pr. I do not think that it is a scope of this pr.